JSTOR-Elasticsearch-with-embeddings.py

```diff
@@ -0,0 +1,540 @@
+        article['front']['article-meta'] = self._process__article_meta(article['front']['article-meta'])
+
+        self._process_if_avail(article['front'], 'notes', lambda v: v if isinstance(v, list) else [v])
+        self._process_if_avail(article.get('back', {}), 'sec', lambda v: v if isinstance(v, list) else [v])
+
+        article.update(article.pop('front'))
+
+        try:
+            article.update(article.pop('back'))
+        except KeyError:
+            pass
+
+        return article
+
+class TXTParser(object):
+    def parse(self, txt_path):
+        txt_root = ET.parse(txt_path).getroot()
+        print(txt_root.tag)
+        if txt_root.tag == 'plain_text':
+            page_seq = [(p.attrib['sequence'], p.text) for p in list(txt_root)]
+            page_seq.sort(key=lambda x: x[0])
+            plain_text_pages = [p for s, p in page_seq]
+            return {'plain_text': str(plain_text_pages)}
+        elif txt_root.tag == 'body':
+            return {'body': str(ET.tostring(txt_root, encoding='utf-8', method='text'))}
+
+
+class EmbeddingParser(object):
+
+    def __init__(self):
+        self.tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')
+        self.model = BertModel.from_pretrained('bert-base-uncased',
+                                  output_hidden_states = True, # Whether the model returns all hidden-states.
+                                  )
+        # Put the model in "evaluation" mode, meaning feed-forward operation.
+        self.model.eval()
+        self.model.to('cuda')
+
+    ##change to single precision or half
+    def word_to_vec(self, text, word_type):
+        marked_text = "[CLS] " + text + " [SEP]"
+
+        # Split the sentence into tokens.
+        tokenized_text = self.tokenizer.tokenize(marked_text)
+        indexed_tokens = self.tokenizer.convert_tokens_to_ids(tokenized_text)
+
+        segments_ids = [1] * len(tokenized_text)
+        tokens_tensor = torch.tensor([indexed_tokens]).to('cuda')
+        segments_tensors = torch.tensor([segments_ids]).to('cuda')
+        #logger.debug("Sentence: {}".format(marked_text))
+        #logger.debug("Length of tokens_tensor {} and segments_tensor {} ".format(
+             #str(len(indexed_tokens)), str(len(segments_ids))))
+
+        with torch.no_grad():
+            outputs = self.model(tokens_tensor, segments_tensors)
+            hidden_states = outputs[2]
+
+        token_embeddings = torch.stack(hidden_states, dim=0)
+        token_embeddings = torch.squeeze(token_embeddings, dim=1)
+        token_embeddings = token_embeddings.permute(1,0,2)
+
+        token_vecs_sum = []
+        for token in token_embeddings:
+            sum_vec = torch.sum(token[-4:], dim=0)
+            token_vecs_sum.append(sum_vec)
+
+        #change later to handle exception
+        target_index = 0
+
+        for i, token_str in enumerate(tokenized_text):
+            if token_str == word_type.value:
+                target_index = i
+                break
```

Comment:
Why have the return statement here and get rid of the break? Does the function need to return zero if nothing is found or can it just not return anything?